spec/src/main/asciidoc/core/implementation.asciidoc

```diff
@@ -51,6 +51,8 @@ An application can prevent packages being split across jars by sealing the packa
 
 The unrestricted set of bean types for a managed bean contains the bean class, every superclass and all interfaces it implements directly or indirectly.
 
+If the resulting set of bean types for a managed bean contains <<legal_bean_types,non-legal bean types>>, these types are removed from the set of bean types.
```

Comment:
`non-legal` - it does not really make sense here. The opposite of legal is `illegal`.
Also the word illegal is already used in spec.

Or, if you don't like the word `illegal`, we may twist the sentence like this:
`The resulting set of bean types for a managed bean consist only of <<legal_bean_types,legal bean types>>, all other types are removed from the set of bean types.`